src/main/java/org/folio/edge/inventory/service/EcsInventoryService.java

```diff
@@ -0,0 +1,121 @@
+    var instanceIds = new ArrayList<String>();
+    instanceView.withArray(INSTANCES).elements()
+        .forEachRemaining((instance) -> instanceIds.add(instance.get(INSTANCE_ID).asText()));
+    return instanceIds;
+  }
+
+  private Map<String, List<String>> getHoldingsTenantsMap(List<String> instanceIds) {
+    var tenantInstanceMap = new HashMap<String, List<String>>();
+    var instanceTenants = getInstanceTenants(instanceIds);
+    instanceTenants.forEach(instanceTenant -> instanceTenant.tenantIds()
+        .forEach(
+            tenantId -> {
+              tenantInstanceMap.putIfAbsent(tenantId, new ArrayList<>());
+              tenantInstanceMap.get(tenantId).add(instanceTenant.instanceId());
+            }));
+    return tenantInstanceMap;
+  }
+
+  private List<JsonNode> getHoldingsAndItemsFromMemberTenants(Map<String, List<String>> tenantInstanceMap) {
+    var context = (FolioExecutionContext) folioExecutionContext.getInstance();
+    return tenantInstanceMap.entrySet().parallelStream()
+        .map(tenantInstances -> context.execute(() -> jsonConverter
+            .readAsTree(inventoryClient.getInventoryViewInstances(
+                getRequestQueryParams(tenantInstances.getValue()), tenantInstances.getKey()))))
+        .toList();
+  }
+
+  private void mergeInstanceViews(JsonNode instanceView1, JsonNode instanceView2) {
+    var instances1 = instanceView1.withArrayProperty(INSTANCES);
+    var instances2 = instanceView2.withArrayProperty(INSTANCES);
+    var instance2Elements = instances2.elements();
+    while (instance2Elements.hasNext()) {
+      var instance2 = instance2Elements.next();
+      var instance1 = JsonNodeUtil.findNodeInArrayWith(instances1, INSTANCE_ID, instance2.get(INSTANCE_ID).asText());
+      instance1.withArrayProperty(HOLDINGS_RECORDS).addAll(instance2.withArrayProperty(HOLDINGS_RECORDS));
+      instance1.withArrayProperty(ITEMS).addAll(instance2.withArrayProperty(ITEMS));
+    }
+  }
+
+  private RequestQueryParameters getRequestQueryParams(List<String> instanceIds) {
+    return new RequestQueryParameters().query(QueryUtil.exactMatchAny(instanceIds, ID)).limit(instanceIds.size());
+  }
+
+  private List<InstanceTenants> getInstanceTenants(List<String> instanceIds) {
+    var context = (FolioExecutionContext) folioExecutionContext.getInstance();
+    return instanceIds.parallelStream()
+        .map(instanceId -> context.execute(() -> {
+          var response = searchClient.getInstanceFacet(FACET, "id==" + instanceId);
+          if (response.getFacets().getHoldingsTenantId().getTotalRecords() == 0) {
+            return new InstanceTenants(instanceId, Collections.EMPTY_LIST);
+          }
+          var tenantIds = response.getFacets().getHoldingsTenantId().getValues()
+              .stream().map(value -> value.getId()).toList();
+          return new InstanceTenants(instanceId, tenantIds);
+        })).toList();
+  }
+
+  public boolean isCentralTenant(String tenantId) {
```

Comment:
Please move public method to the top of the class

Reply:
moved